### ISSUE TITLE
Something weird with the Level Meter UI

### ISSUE THREAD

**Issue by Digitalone1** [wwmm/easyeffects#2342]
Why is there a thin line at the bottom of the Loudness box list?

![Schermata del 2023-05-14 20-37-26](https://github.com/wwmm/easyeffects/assets/25790525/26fd07aa-31f2-4925-bda7-f1d653e94fc2)

It's the only box showing it. There should be something weird with the layout of the widgets inside.

**Comment by wwmm** [comment]
I did not notice this. Very strange. In other to have all the GtkScale with the same size I had to use a GtkListBox instead of the adwaita widget. But this isn't a reason for something so weird.

**Comment by wwmm** [comment]
If I set `show-separators` to false that bottom line disappears. But then we do not have the one between the lines... I wonder if there is a way to make the title of each `AdwActionRow` to have the same size and position. This way it would probably not be necessary to use a GtkListbox and applying css classes to make it look like the adwaita widget....

**Comment by wwmm** [comment]
>But then we do not have the one between the rows...

Or maybe we can take the lazy approach and let this box without lines between rows and pretending it is a style choice :smile:

**Comment by Digitalone1** [comment]
> > But then we do not have the one between the rows...
> 
> Or maybe we can take the lazy approach and let this box without lines between rows and pretending it is a style choice 😄

Then it would be easier to set a grid with 3 columns inside a box...

**Comment by wwmm** [comment]
>Then it would be easier to set a grid with 3 columns inside a box...

I am not sure if the grid will follow the `rich-list` class. The `card` class probably... If it is not going to look like a list it is better to keep the listbox without lines.

**Comment by Digitalone1** [comment]
> If I set `show-separators` to false that bottom line disappears. But then we do not have the one between the rows...

So it's the separator shown on the last row which should not be shown. Maybe a GTK bug, but now that I think about it, the two boxes have separators with different colors. It's not really consistent.

**Comment by wwmm** [comment]
>the two boxes have separators with different colors

Different colors?

**Comment by Digitalone1** [comment]
> If it is not going to look like a list it is better to keep the listbox without lines.

Try it. And another thing. For the peak measurement, isn't it better to use `Maximum True Peak` or `Max True Peak`?

**Comment by Digitalone1** [comment]
> > the two boxes have separators with different colors
> 
> Different colors?

Yes, look here:

![image](https://github.com/wwmm/easyeffects/assets/25790525/b0b97431-c3e4-4b0a-857e-243ddcae6236)

The left box has a transparent separator. Maybe it's different on the light theme?

**Comment by wwmm** [comment]
>For the peak measurement, isn't it better to use Maximum True Peak or Max True Peak?

It is a more precise description. But I was not sure about how a longer title would look like at that location. Specially translated. But I think it is fine to change to this new title.

>Try it.

The annoying side of the GtkGrid is that we have to specify each element row and columns. Once you decide to change the position of one element you may have to edit the indices of many others. And from a performance point of view I doubt the grid is better than the listbox. So there isn't really an advantage.

>The left box has a transparent separator. Maybe it's different on the light theme?

Oh... It is definitely different on a light theme. I did not think about testing in the dark theme.

**Comment by Digitalone1** [comment]
> It is a more precise description. But I was not sure about how a longer title would look like at that location. Specially translated. But I think it is fine to change to this new title.

Yes, I mean there are many peaks on the signal, but we are showing the maximum. I say it for beginners that may not understand...

**Comment by wwmm** [comment]
Ideally I would rather have used an `AdwActionRow` inside a `AdwPreferencesPage`. But I could not find a way to make GtkScale located after titles with different lengths to have the same size. Really annoying...

**Comment by wwmm** [comment]
At this moment I am replacing `rubberband` by `soundtouch` in the pitch plugin. I will try to take a look at the level meter again after this.

**Comment by Digitalone1** [comment]
These are two bugs that should be reported to GTK. The show separator property is showing a line on the last row and separators are different between light and dark mode.

> At this moment I am replacing `rubberband` by `soundtouch` in the pitch plugin. I will try to take a look at the level meter again after this.

No problem. Another thing to address is that the level meter can be placed at the last position of the pipeline. At the moment we place effects at the second to last position if there are limiters plugins at the bottom. But we can do an exception for the level meter (as we do with limiters).

**Comment by wwmm** [comment]
>These are two bugs that should be reported to GTK. The show separator property is showing a line on the last row and separators are different between light and dark mode.

I am not sure it is a bug. Looking at the listbox with no class applied I think that what gtk does is showing a horizontal line after each line. It just happens it does not do any special handling for the last line.  And the difference in color may be just adwaita applied a different style to the AdwActionRow. The real problem here is having to use different containers just because Adwaita did not obey any rule I tried to make the GtkScale to have the same length across rows.

>No problem. Another thing to address is that the level meter can be placed at the last position of the pipeline (now it's not always happening). At the moment we place effects at the second to last position if there are limiters plugins at the bottom. But we can do an exception when adding the level meter (as we do with limiters).

>I will talk about this in another ticket.

Ok. I think it is fine to handle the level meter like the other plugins. It isn't on the same level as the limiter because it won't protect anything.

**Comment by Digitalone1** [comment]
I suggest to do like this in the AutoGain:

![image](https://github.com/wwmm/easyeffects/assets/25790525/318d5ffe-0292-4ab0-aa9e-5b934f7c07a4)

And report the Left and Right Max Peak as done in the Gates/Compressors level meters:

![image](https://github.com/wwmm/easyeffects/assets/25790525/287ac450-f669-467a-bb50-0fcf537fd1d2)

Just above the input/output Level Meters.

**Comment by wwmm** [comment]
>I suggest to do like this in the AutoGain:

It will work. It is just that I am a little tired of the current autogain layout and wanted to try something new :smile:. Now that the migration to soundtouch is finished I will investigate a little more about a solution to the level meter.

**Comment by wwmm** [comment]
Problem fixed in the latest master
![Screenshot from 2023-05-18 11-49-09](https://github.com/wwmm/easyeffects/assets/16225172/509e2aab-0efa-44aa-9e28-a3997f7c4b96)
Using a `AdwPreferencesRow` was the answer. It is the base class of `AdwActionRow`. 

I think we can close this issue.

**Comment by Digitalone1** [comment]
Very nice!
